aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java

```diff
@@ -143,6 +142,7 @@ void initialize(
     this.closeableGroup = new CloseableGroup();
     closeableGroup.addCloseable(dynamo);
     closeableGroup.addCloseable(fileIO);
+    closeableGroup.addCloseable(metricsReporter());
```

Comment:
I just wanted to point out that this effectively will initialize the reporter much earlier (during catalog instantiation) than previously (during table loading). This is probably ok